contracts/sfc/SFC.sol

```diff
@@ -817,12 +820,13 @@ contract SFC is OwnableUpgradeable, UUPSUpgradeable, Version {
         return rewards;
     }
 
-    /// Burn FTM tokens.
+    /// Burn native tokens.
     /// The tokens are sent to the zero address.
-    function _burnFTM(uint256 amount) internal {
-        if (amount != 0) {
+    function _burnNativeTokens(uint256 amount) internal {
+        if (amount != 0 && totalSupply >= amount) {
```

Comment:
if `amount > totalSupply` you could also do `amount = totalSupply;` and then burn that amount